### ISSUE TITLE
Upgraded to 2.11.2: "Failedto migrate the DB... Dirty database version"

### ISSUE THREAD

**Issue by dadlerj** [sourcegraph/issues#148]
- **Sourcegraph version:** self-hosted
- **Platform:** ?

#### Steps to reproduce:

A user contacted us to report that they received the error message below when upgrading to 2.11.2. I've asked the user to update this issue with more information, including their previous Sourcegraph release. 

#### Expected: 

2.11.2 to not have any DB migration issues

#### Actual:

(message from user)

>Hello,
>
>I just tried to upgrade Sourcegraph to 2.11.2.
>I'm getting this error:

```
09:39:27 frontend | Failedto migrate the DB. Please contact support@sourcegraph.com for further assistance: Dirty database version 1528395535. Fix and force version.
09:39:27 frontend | Terminating frontend
frontend died. Shutting down...
```

>Tried redis-check-aof and redis-check-rdb, did not see any errors.
>
>Please tell me how to fix this!
>Thanks,

**Comment by dloxtermann** [comment]
Hi there,
I submitted this Issue.
I'm running Sourcegraph on Docker 1.13.1 / Ubuntu 16.04.4 LTS.

I think i was running Sourcegraph 2.9. before that.
I also tried running version 2.9, which gave me the same error message.

Thanks,
Daniel

**Comment by nicksnyder** [comment]
It looks like a migration failed at some point which has set a dirty bit in your database. We did identify a race condition in the migration tool that we used in 2.9 and fixed it in 2.10, but this doesn't help you if your database is already in an inconsistent state.

The least hassle way to get a working instance is to probably configure a fresh instance (e.g. delete your previous data). If you have data that you want to save, then we will need to run some sql commands to figure out where the migration failed and how to recover.
